internal/db/db.go

```diff
@@ -34,9 +37,42 @@ func StringToDbType(dialect string) (DbType, error) {
 	}
 }
 
-// Open a database connection which is long-lived.
-// You need to call Close() on the returned gorm.DB
-func Open(dbType DbType, connectionUrl string) (*gorm.DB, error) {
+// DB is a wrapper around the ORM
+type DB struct {
+	*gorm.DB
+}
+
+// SqlDB returns the underlying sql.DB
+//
+// Note: This func is not named DB(), because our choice of ORM (gorm) which is
+// embedded for various reasons, already exports a DB() func and it's base type
+// is also an export DB type
+func (d *DB) SqlDB(ctx context.Context) (*sql.DB, error) {
+	const op = "db.(DB).SqlDB"
+	if d.DB == nil {
+		return nil, errors.New(ctx, errors.Internal, op, "missing underlying database")
+	}
+	return d.DB.DB()
+}
+
+// Close the underlying sql.DB
+func (d *DB) Close(ctx context.Context) error {
+	const op = "db.(DB).Close"
+	if d.DB == nil {
+		return errors.New(ctx, errors.Internal, op, "missing underlying database")
+	}
+	underlying, err := d.DB.DB()
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	return underlying.Close()
+}
+
+// Open a database connection which is long-lived. The options of
+// WithGormFormatter and WithMaxOpenConnections are supported.
+//
+// Note: you need to call Close() on the returned DB
```

Comment:
interestingly, the `sql` docs for `Close` state:

> It is rare to Close a DB, as the DB handle is meant to be long-lived and shared between many goroutines.

And for `Open`:

> It is rarely necessary to close a DB.

And I think the only spot we are calling Close (outside of tests) in is `DestroyDevDatabase` to ensure the connections are closed before destroying the database.

Reply:
updated comment/doc